### ISSUE TITLE
Immich - Insufficient amd.com/gpu

### ISSUE THREAD

**Issue by Clasyc** [truenas/charts#2336]
I have **AMD Ryzen 4650G** CPU.

I want to use hardware acceleration for transcoding. If I set to use AMD GPU in configuration when deploying Immich app, it will never finish deployment because of 

```
Events:
  Type     Reason            Age                  From               Message
  ----     ------            ----                 ----               -------
  Warning  FailedScheduling  29m                  default-scheduler  0/1 nodes are available: 1 Insufficient amd.com/gpu. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
  ```
  
  ```
  Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests    Limits
  --------           --------    ------
  cpu                290m (2%)   18 (150%)
  memory             570Mi (1%)  35938Mi (114%)
  ephemeral-storage  0 (0%)      0 (0%)
  hugepages-1Gi      0 (0%)      0 (0%)
  hugepages-2Mi      0 (0%)      0 (0%)
  amd.com/gpu        1           1
Events:              <none>
```

It appears that the `immich-machinelearning` pod is already utilizing one AMD GPU allocation, preventing the `immich-microservices` pod from accessing it.

What should I do to resolve this?

**Comment by stavros-k** [comment]
Hello, isn't the GPU used by both containers? One for transcoding and one for machine learning.
In that case you need to have available "2" resources.

But I think AMD does not support slicing/sharing, which might require changes to account for that.

**Comment by Clasyc** [comment]
>Hello, isn't the GPU used by both containers? One for transcoding and one for machine learning.

Yes, that is my desired way of using it. I want to use the same GPU for both containers.

However, I'm not sure where I should attempt to adjust this. Should I explore the AMD GPU Kube plugin implementation, or should I look elsewhere?

**Comment by Clasyc** [comment]
I might be wrong, but can Helm charts be updated to only use the GPU for immich-microservices and not for machinelearning (optional), since Immich machine learning [doesn't support AMD GPUs](https://immich.app/docs/features/ml-hardware-acceleration)?

**Comment by stavros-k** [comment]
> I might be wrong, but can Helm charts be updated to only use the GPU for immich-microservices and not for machinelearning (optional), since Immich machine learning [doesn't support AMD GPUs](https://immich.app/docs/features/ml-hardware-acceleration)?

Yes, that's the idea! But it won't be only for AMD will be for any gpu.
Will look into that, as it will need a small migration too.

**Comment by tannisroot** [comment]
> I don't think k8s amd plugin allows for slicing at all. I'll have to investigate that

Does Intel's allow slicing?
I want to do the same thing as OP does but I'm waiting for Dragonfish release for DG2 support :)

**Comment by stavros-k** [comment]
> > I don't think k8s amd plugin allows for slicing at all. I'll have to investigate that
> 
> Does Intel's allow slicing? I want to do the same thing as OP does but I'm waiting for Dragonfish release for DG2 support :)

Yes Intel allows slicing afaik!

**Comment by hrz6976** [comment]
Got the same issue with a NVIDIA P40 24GB GPU on latest stable (24.04.0): `2024-05-08 21:21:48 0/1 nodes are available: 1 Insufficient nvidia.com/gpu.`

**Comment by hrz6976** [comment]
For anyone come across this issue: I find unchecking use GPU for microservices does the job for me. 
Another trap: the postgres container requires a data folder owned by user id 1000 and permission 700.
